### PR TITLE
Bind tensor derivatives in Python

### PR DESCRIPTION
**Pull request by nilsvu**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/DataStructures/Tensor/Python/Tensor.cpp [hunk 1]
```diff
@@ -155,8 +156,10 @@ void bind_tensor(py::module& m) {
...
 
-#define INSTANTIATE(_, data) \
-  bind_tensor_impl<tnsr::I<DTYPE(data), DIM(data), FRAME(data)>>(m, "I");
+#define INSTANTIATE(_, data)                                                \
+  bind_tensor_impl<tnsr::I<DTYPE(data), DIM(data), FRAME(data)>>(m, "I");   \
+  bind_tensor_impl<tnsr::ii<DTYPE(data), DIM(data), FRAME(data)>>(m, "ii"); \
+  bind_tensor_impl<tnsr::II<DTYPE(data), DIM(data), FRAME(data)>>(m, "II");
 
   GENERATE_INSTANTIATIONS(INSTANTIATE, (double, DataVector), (1, 2, 3),
                           (Frame::ElementLogical, Frame::Inertial))
```
Comment (nilsdeppe) [inline src/DataStructures/Tensor/Python/Tensor.cpp]: Should we add a final list `(I, ii, II)`? That seems more scalable and would make it relatively easy to just list all the things in `TypeAliases.hpp`

File: src/DataStructures/Tensor/Python/Tensor.cpp [hunk 2]
```diff
@@ -34,21 +34,29 @@ std::string dtype_to_name() {
...
-  if constexpr (TensorType::rank() == 0) {
+  if constexpr (Kind == TensorKind::Scalar) {
     return name + dtype_to_name<typename TensorType::type>();
-  } else {
+  } else if constexpr (Kind == TensorKind::Tnsr) {
     return "Tensor" + name + dtype_to_name<typename TensorType::type>() +
            std::to_string(TensorType::index_dim(0)) +
            get_output(get<0>(TensorType::index_frames()));
+  } else if constexpr (Kind == TensorKind::Jacobian) {
+    return name + dtype_to_name<typename TensorType::type>() +
```
Comment (nilsdeppe) [inline src/DataStructures/Tensor/Python/Tensor.cpp:51]: Can you add a comment explaining why Jacobians are handled specially?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: src/DataStructures/Tensor/Python/Tensor.cpp (hunk 1 above)
Comment (nilsvu) [reply src/DataStructures/Tensor/Python/Tensor.cpp]: The "I" has to go into the string literal passed as `name`. This doesn't work: `"TENSOR(data)"` where `#define TENSOR(data) BOOST_PP_TUPLE_ELEM(3, data)`. Is this easy to fix?

Comment (nilsdeppe) [reply src/DataStructures/Tensor/Python/Tensor.cpp]: https://stackoverflow.com/questions/13741048/return-quoted-string-in-c-macro maybe `BOOST_PP_STRINGIZE`?

Comment (nilsvu) [reply src/DataStructures/Tensor/Python/Tensor.cpp]: works, thanks! Pushed fixup

**Review by nilsdeppe — DISMISSED** [review]
:tada: looks great! Thanks for doing all this!

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**